test/generators/go/presets/__snapshots__/JsonTagPreset.spec.ts.snap

```diff
@@ -0,0 +1,49 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`JSON preset should render serialize and deserialize code 1`] = `
+"// Test represents a Test model.
+type Test struct {
+  StringProp string \`json:\\"string prop,omitempty\\"\`
+  NumberProp float64 \`json:\\"numberProp,omitempty\\"\`
+  EnumProp *EnumTest \`json:\\"enumProp,omitempty\\"\`
+  ObjectProp *NestedTest \`json:\\"objectProp,omitempty\\"\`
+  AdditionalProperties map[string]string \`json:\\"additionalProperties,omitempty\\"\`
+}"
+`;
+
+exports[`JSON preset should render serialize and deserialize code 2`] = `
+"// EnumTest represents an enum of EnumTest.
+type EnumTest uint
+
+const (
+  EnumTestSomeSpaceEnumSpaceString EnumTest = iota
+  EnumTestTrue
+  EnumTestCurlyleftQuotationTestQuotationColonQuotationTestQuotationCurlyright
+  EnumTestNumber_2
+)
+
+// Value returns the value of the enum.
+func (op EnumTest) Value() any {
+	if op >= EnumTest(len(EnumTestValues)) {
+		return nil
+	}
+	return EnumTestValues[op]
+}
```

Comment:
You can have even enums as string, so I think that we should somehow base logic on type of enum.

https://www.sohamkamani.com/golang/enums/

cc @smoya

Reply:
Pretty sure, without fully remembering it, was this not the desired behaviour we sketched out Sergio?

Reply:
hmm, I don't remember, cc @smoya

Reply:
Okay, the implementation is insufficient to support encoding and decoding enums and structs into and from JSON. Gonna make some alterations and re-request a review once done 🙂

Reply:
@magicmatatjahu what about supporting multiple types inside those enum? Not all values will be `string`

Reply:
But it's a limitation of enums in languages, even in TS you can have string or number values. In this case you can define enum values as string or numbers. That's my point.

Reply:
> But it's a limitation of enums in languages, even in TS you can have string or number values. In this case you can define enum values as string or numbers. That's my point.

There is one missing piece @jonaslagoni didn't add to this code but it was in the implementation we baked together:

```go
func (op *EnumTest) UnmarshalJSON(raw []byte) error {
	var v any
	if err := json.Unmarshal(raw, &v); err != nil {
		return err
	}
	*op = ValuesToEnumTest[v]
	return nil
}
```

With that, when unmarshaling, the value will be transformed to (and only to) the `uint` const.
There is a case that won't work, and its when the value is an object, because that `ValuesToEnumTest` map can't have a key that is a map. However I'm gonna give another though to it.